src/agent.rs

```diff
@@ -20,30 +19,36 @@ pub struct Agent {
     path: String,
     args: Vec<String>,
     pub socket: TcpStream,
-    child: Receiver<i32>,
+    child: Receiver<Output>,
     pub alive: bool,
     exit_value: Option<ExitStatus>,
 }
 
+fn cipher_string_to_ossl(input: &str) -> String{
+    String::from(input).replace("TLS_", "").replace("AES_", "AES").replace("_", "-")
+        .replace("-WITH", "")
+}
 
 impl Agent {
     pub fn new(name: &str,
                path: &str,
                agent: &Option<TestCaseAgent>,
                args: Vec<String>,
-               conf: &TestConfig)
+               ipv4: bool)
                -> Result<Agent, i32> {
         // IPv6 listener by default, IPv4 fallback, unless IPv4 is forced.
         let addr6 = "[::1]:0".parse().unwrap();
         let addr4 = "127.0.0.1:0".parse().unwrap();
-        let listener = match conf.force_ipv4 {
+        let listener = match ipv4 {
             false => TcpListener::bind(&addr6).or_else(|_| {
                 TcpListener::bind(&addr4)}).unwrap(),
             true => TcpListener::bind(&addr4).unwrap(),
         };
 
+        let ossl_cipher_format = path.contains("bssl_shim") || path.contains("ossl_shim");
```

Comment:
What if those names change at some point? Could there be a better way to detect this?

Reply:
I see no obvious way to make this more robust at the moment, since interop is otherwise unaware of what kind of shim it is talking to at this point. I will create an issue to improve this later, if that's fine for you.